code/drasil-gool/GOOL/Drasil/LanguageRenderer/CLike.hs

```diff
@@ -54,20 +54,20 @@ charRender = "char"
 voidRender = "void"
 
 float :: (RenderSym r) => VSType r
-float = toState $ typeFromData Float floatRender (text floatRender)
+float = typeFromData Float floatRender (text floatRender)
 
 double :: (RenderSym r) => VSType r
-double = toState $ typeFromData Double doubleRender (text doubleRender)
+double = typeFromData Double doubleRender (text doubleRender)
 
 char :: (RenderSym r) => VSType r
-char = toState $ typeFromData Char charRender (text charRender)
+char = typeFromData Char charRender (text charRender)
 
 listType :: (RenderSym r) => String -> VSType r -> VSType r
-listType lst = onStateValue (\t -> typeFromData (List (getType t)) (lst 
-  `containing` getTypeString t) (text lst <> angles (RC.type' t)))
-  
+listType lst = on1StateWrapped (\t -> typeFromData (List (getType t)) (lst 
```

Comment:
I think I would prefer to use `do` notation instead of `on1StateWrapped`, everywhere.

Reply:
Like here. There were reasons before to provide combinators, but because of various refactors, these have gone away, so that now the code would be clearer using `do` notation.